cpp/src/neighbors/detail/vamana/vamana_search.cuh

```diff
@@ -0,0 +1,283 @@
+/*
```

Comment:
Just a nit pick. We might be better off calling this `greedy_search.cuh`. I believe the DiskANN paper talks about "Vamana" as the indexing algorithm alone, and the search part is just greedy search. And we will also need the same functions for DiskANN search?